src/Common/src/TypeSystem/Ecma/EcmaType.cs

```diff
@@ -42,6 +43,19 @@ internal EcmaType(EcmaModule module, TypeDefinitionHandle handle)
 #endif
         }
 
+        public override int GetHashCode()
+        {
```

Comment:
The hashcode should be cached.  (To make the caching simple, it should be ok to recompute it again and again in the super rare case of hashcode == 0.)